src/vdom/diff.js

```diff
@@ -94,18 +94,20 @@ function idiff(dom, vnode, context, mountAll, componentRoot) {
 
 
 	// If the VNode represents a Component, perform a component diff:
-	if (typeof vnode.nodeName==='function') {
+	let vnodeName = vnode.nodeName;
+	if (typeof vnodeName==='function') {
 		return buildComponentFromVNode(dom, vnode, context, mountAll);
 	}
 
 
 	// Tracks entering and exiting SVG namespace when descending through the tree.
-	isSvgMode = vnode.nodeName==='svg' ? true : vnode.nodeName==='foreignObject' ? false : isSvgMode;
+	isSvgMode = vnodeName==='svg' ? true : vnodeName==='foreignObject' ? false : isSvgMode;
 
 
 	// If there's no existing element or it's the wrong type, create a new one:
-	if (!dom || !isNamedNode(dom, String(vnode.nodeName))) {
```

Comment:
Looking at the history, the purpose of the `String` constructor here is to turn `undefined` into `"undefined"` (see 4d6b20d5e849d9dd4b5c6fb9ff9f23bc3f05ebba). That could potentially be moved into the "h" function in future so that everywhere else can rely on `vnode.nodeName`'s type being `Function|string`.